Microsoft/helium
# add types for resitfy and body-parser, update typescript

This fixes dependency issues currently in master preventing build.

Needed types added to the `package.json` for `restify` and `body-parser`.

Also updated the typescript version to get around errors with these types.

Also reorders deps into dependencies and devDependencies.

To test:

1. clone this code
2. `npm install`
3. `npm run build`

should see no errors related to dependencies
